robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -1,191 +1,559 @@
+    // setup a subscriber for the rover_position topic
+    let rover_position_listener = new ROSLIB.Topic({
+        ros: ros,
+        name: 'rover_position',
+        messageType: 'geometry_msgs/Point'
+    })
+    rover_position_listener.subscribe(function(message) {
+        $('#rover-latitude').text(message.x)
+        $('#rover-longitude').text(message.y)
+        $('#rover-heading').text(message.z)
+    })
+    // setup a subscriber for the antenna_goal topic
+    let antenna_goal_listener = new ROSLIB.Topic({
+        name: 'antenna_goal',
+        messageType: 'geometry_msgs/Point',
+        ros: ros
+    })
+    antenna_goal_listener.subscribe(function(message) {
+        $('#antenna-stats-rec-angle').text(parseFloat(message.x).toFixed(3))
+        $('#antenna-stats-dist-to-rover').text(parseFloat(message.y).toFixed(2))
+    })
+    // setup topics to communicate with GoalsNode
+    let create_goal_publisher = new ROSLIB.Topic({
+        ros: ros,
+        name: 'create_goal',
+        messageType: 'mcu_control/RoverGoal'
+    })
+    let delete_goal_publisher = new ROSLIB.Topic({
+        ros: ros,
+        name: 'delete_goal',
+        messageType: 'std_msgs/String'
+    })
+    goal_list_subscriber.subscribe(function(message) {
+        goalList = message.goal_list
+        if (count != goalList.length) {
+            count = goalList.length
+
+        }
+        if (count > 0) {
+            goal_latitude.set(goalList[0].lat)
+            goal_longitude.set(goalList[0].long)
+            $('#goal-stats-lat').text(goalList[0].lat.toFixed(6))
+            $('#goal-stats-long').text(goalList[0].long.toFixed(6))
+        } else {
+            goal_latitude.set(false)
+            goal_longitude.set(false)
+            $('#goal-stats-lat').text('----')
+            $('#goal-stats-long').text('----')
+        }
+
+        // this is a meme implementation of a loop to update the params without resorting to a message type + subscriber + publisher
+        updateAntennaParams()
+    })
+    // setup gps parameters for antenna directing
+    let antenna_latitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'antenna_latitude'
+    })
+    let antenna_longitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'antenna_longitude'
+    })
+    let antenna_start_dir = new ROSLIB.Param({
+        ros: ros,
+        name: 'antenna_start_dir'
+    })
+    let has_gps_goal = new ROSLIB.Param({
+        ros: ros,
+        name: 'has_gps_goal'
+    })
+    // setup gps parameters for rover goals
+    let goal_latitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'goal_latitude'
+    })
+    let goal_longitude = new ROSLIB.Param({
+        ros: ros,
+        name: 'goal_longitude'
+    })
+    // setup a subscriber for the rover_goal topic
+    let rover_goal_listener = new ROSLIB.Topic({
+        ros: ros,
+        name: 'rover_goal',
+        messageType: 'geometry_msgs/Point'
+    })
+
+    goalList = []
+    count = 0;
```

Comment:
what count? The variable should have a better name. Also if its the count of goals you can just get the length of the goal list array